apps/room/tests/test_services/test_room_list_ordering.py

```
@@ -0,0 +1,94 @@
+    def test_room_with_transaction_uses_transaction_timestamp(self, room, user):
+        transaction = ParentTransactionFactory(room=room, paid_by=user, currency=room.preferred_currency)
+        expected_ts = timezone.now() - timedelta(days=1)
+        from apps.transaction.models import ParentTransaction
+
+        ParentTransaction.objects.filter(pk=transaction.pk).update(lastmodified_at=expected_ts)
+
+        annotated = Room.objects.filter(pk=room.pk).annotate_last_activity().get()
+        # Should be close to expected_ts (within a second for DB rounding)
+        assert abs((annotated.last_activity - expected_ts).total_seconds()) < 2
+
+    def test_room_without_transactions_falls_back_to_room_lastmodified_at(self, db):
```

Ruff (0.15.10)

[warning] Unused method argument: `db`

(ARG002)